test/unit/lib/serverless.test.js

```diff
@@ -174,4 +176,151 @@ describe('test/unit/lib/serverless.test.js', () => {
       expect(serverless.config).to.have.property('servicePath');
     });
   });
+
+  describe('Extend configuration', () => {
+    let serverless;
+    let parseEntriesStub;
+
+    before(async () => {
+      parseEntriesStub = sinon.stub().returns(new Map([]));
+      try {
+        await runServerless({
+          noService: true,
+          command: 'info',
+          modulesCacheStub: {
+            './lib/configuration/variables/resolve-meta': {
+              parseEntries: parseEntriesStub,
+            },
+          },
+          hooks: {
+            beforeInstanceInit: (serverlessInstance) => {
```

Comment:
No need to do that.
- `serverless` is returned from `runServerless`
- You can safely run `package` command without side effects

Reply:
As the function should not be callable after init, i have to break before init(). Otherwise it will allways throw and i can't test this feature.

Reply:
Where should the test be created?
It would be more fitting to have it in the integration folder instead of unit.

Reply:
Integration tests we have configured depend on AWS infra. While tests configured with `runServerless` do not depend on any external resources and can be run offline.

See the linked fixture. We also already have some tests configured that run against that fixture, e.g. https://github.com/serverless/serverless/blob/c79737efe49a1484bee4c3e64d512d824a70824e/test/unit/lib/classes/plugin-manager.test.js#L1596-L1600